### PR TITLE
Add support for Argonaut

### PR DESCRIPTION
**Pull request by alonsodomin**
Support for encoding and decoding values using Argonaut.

### CODE REVIEW

File: enumeratum-argonaut/src/main/scala/enumeratum/values/ArgonautValueEnum.scala [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  implicit val argonautDecoder = Argonauter.decoder(this)
+}
+
+/**
+  * ArgonautEnum for ByteEnumEntry
+  */
+trait ByteArgonautEnum[EntryType <: ByteEnumEntry] extends ArgonautValueEnum[Byte, EntryType] {
+  this: ValueEnum[Byte, EntryType] =>
+
+  implicit val argonautEncoder = Argonauter.encoder(this)
```
Comment (lloydmeta) [inline enumeratum-argonaut/src/main/scala/enumeratum/values/ArgonautValueEnum.scala]: [Looks like](https://travis-ci.org/lloydmeta/enumeratum/builds/167795213#L923) the compiler can't find the `EncodeJson[Byte]` implicit declared inside the package object. I'm not exactly sure why; could be something to do with the fact that this file comes first alphabetically ? 😨

File: enumeratum-argonaut/src/test/scala/enumeratum/TrafficLight.scala [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+/**
+  * Created by alonsodomin on 14/10/2016.
+  */
+sealed trait TrafficLight extends EnumEntry
+object TrafficLight extends Enum[TrafficLight] with ArgonautEnum[TrafficLight] {
+  case object Red    extends TrafficLight
+  case object Yellow extends TrafficLight
+  case object Green  extends TrafficLight
+  override def values: Seq[TrafficLight] = Seq(Red, Yellow, Green)
```
Comment (lloydmeta) [inline enumeratum-argonaut/src/test/scala/enumeratum/TrafficLight.scala]: `val values = findValues` might be nicer here.

**Comment by lloydmeta** [comment]
Thanks for the PR. I especially liked your use of `Short` when decoding `Byte`; I should probably replace that elsewhere too :)

Left a bit of feedback here and there, no rush.

**Comment by alonsodomin** [comment]
That compile error due to missed implicits was a weird one! I could only reproduce it by compiling from scratch in 2.10... the fix apparently is to add type annotations to the encoder/decoder, after that it can resolve the implicit (I finally added the type annotations to all the enum values, for consistency).

btw, for some reason scalafmt was making SBT fail during compile time when working on 2.10, are you familiar with that? I temporarly disabled the formatting to work on this, hence the last "futile" commit to revert it back.

**Comment by lloydmeta** [comment]
@alonsodomin strange, I don't seem to be having issues with scalafmt when working on 2.10.

In any case, everything looks good now ! Thanks a lot !

**Comment by lloydmeta** [comment]
Just pushed `enumeratum-argonaut`, version `1.4.15` to Sonatype releases. It should be up on Maven central in the next hour or so. Thanks again 👍

**Comment by alonsodomin** [comment]
Didn't expect such a quick turnover but that's great, it will help removing some boilerplate in clien projects.

**Pull request merged**